tests/index_routines.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from test_tools.generators import mk_seq_array
+
+import cunumeric as num
+from legate.core import LEGATE_MAX_DIM
+
+
+def test():
+    choices1 = [
+        [0, 1, 2, 3],
+        [10, 11, 12, 13],
+        [20, 21, 22, 23],
+        [30, 31, 32, 33],
+    ]
+    a1 = [2, 3, 1, 0]
+    num_a1 = num.array(a1)
+    num_choices1 = num.array(choices1)
+
+    aout = np.array([2.3, 3.0, 1.2, 0.3])
+    num_aout = num.array(aout)
+
+    assert np.array_equal(
+        np.choose(a1, choices1, out=aout),
+        num.choose(num_a1, num_choices1, out=num_aout),
+    )
```

Comment:
Can you additionally verify that `aout` and `num_aout` are equal? So we can verify that the operation indeed updates the arrays in-place.